### PR TITLE
PLUTO-661 added a system property called test.browser.binary

### PR DESCRIPTION
**Pull request by vsingleton**

### CODE REVIEW

File: portlet-tck_3.0/driver/src/main/java/javax/portlet/tck/driver/TCKSimpleTestDriver.java [hunk 1]
```diff
@@ -228,15 +233,21 @@ public static void setUpBeforeClass() throws Exception {
       System.out.println("   PasswordId   =" + passwordId);
       System.out.println("   Browser      =" + browser);
       System.out.println("   Driver       =" + wd);
+      System.out.println("   binary       =" + binary + " ... used for ChromeDriver & FirefoxDriver only, for now");
 
       if (browser.equalsIgnoreCase("firefox")) {
-         driver = new FirefoxDriver();
+         driver = new FirefoxDriver(new FirefoxBinary(new File(binary)), new FirefoxProfile());
```
Comment (ngriffin7a) [inline portlet-tck_3.0/driver/src/main/java/javax/portlet/tck/driver/TCKSimpleTestDriver.java]: @vsingleton Does this mean that `-Dtest.browser.binary` must *always* be specified? If so, then the old behavior needs to be kept. In other words, need to put in a condition `if (binary == null) ... else`

File: portlet-tck_3.0/driver/src/main/java/javax/portlet/tck/driver/TCKSimpleTestDriver.java [hunk 2]
```diff
@@ -228,15 +233,21 @@ public static void setUpBeforeClass() throws Exception {
...
          driver = new InternetExplorerDriver();
       } else if (browser.equalsIgnoreCase("chrome")) {
          System.setProperty("webdriver.chrome.driver", wd);
-         driver = new ChromeDriver();
+         ChromeOptions options = new ChromeOptions();
+         options.setBinary(binary);
+         options.addArguments("--headless");
+         options.addArguments("--disable-infobars");
+         driver = new ChromeDriver(options);
+         System.out.println("!! option       = headless");
```
Comment (ngriffin7a) [inline portlet-tck_3.0/driver/src/main/java/javax/portlet/tck/driver/TCKSimpleTestDriver.java]: @vsingleton Is there a way to disable *headless* chrome like `-Dintegration.browser.headless=false` in the Liferay Faces Test framework? https://github.com/ngriffin7a/liferay-faces-test-selenium/blob/663c1bd78b53f3746333c5b191d9eec6faabef0c/pom.xml#L105
If not then need to add that.

File: portlet-tck_3.0/driver/src/main/java/javax/portlet/tck/driver/TCKSimpleTestDriver.java (hunk 1 above)
Comment (vsingleton) [reply portlet-tck_3.0/driver/src/main/java/javax/portlet/tck/driver/TCKSimpleTestDriver.java]: Done.

File: portlet-tck_3.0/driver/src/main/java/javax/portlet/tck/driver/TCKSimpleTestDriver.java (hunk 2 above)
Comment (vsingleton) [reply portlet-tck_3.0/driver/src/main/java/javax/portlet/tck/driver/TCKSimpleTestDriver.java]: Done.  Same branch PLUTO-661_01.  Same PR:
https://github.com/ngriffin7a/portals-pluto/pull/9

File: portlet-tck_3.0/driver/src/main/java/javax/portlet/tck/driver/TCKSimpleTestDriver.java (hunk 1 above)
Comment (vsingleton) [reply portlet-tck_3.0/driver/src/main/java/javax/portlet/tck/driver/TCKSimpleTestDriver.java]: Um ... i thought this was done, but I may need to change the property names.  So please wait again on this.

File: portlet-tck_3.0/driver/pom.xml [hunk 3]
```diff
@@ -662,6 +662,14 @@
                            <value>${test.browser.webDriver}</value>
```
Comment (ngriffin7a) [inline portlet-tck_3.0/driver/pom.xml:662]: @vsingleton "test.browser.webDriver" is the property that the Portlet 3.0 TCK already has for specifying a binary. For example, this is hard-coded:
`<test.browser.webDriver>c:/ntutil/chromedriver.exe</test.browser.webDriver>`

Is it necessary to introduce a new property named "test.browser.binary" in order to make this work? Or should we just rely on the developer overriding the value of "test.browser.webDriver" via the command-line?

Comment (ngriffin7a) [reply portlet-tck_3.0/driver/pom.xml:662]: @vsingleton Also need to introduce a property named "test.browser.headless" similar to the "integration.browser.headless" that we have in liferay-faces-test-selenium.

**Comment by ngriffin7a** [comment]
Merged. Thank you. See code review commit https://github.com/apache/portals-pluto/commit/518089566e25805b3aa6a2572b19f3d87ffac4f2

**Pull request closed without merging**